### PR TITLE
Normalize builtin types to dtypes.

### PR DESCRIPTION
**Pull request by ysiraichi**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #106074

Fix: #105052
Follow-up: #105588

This PR normalizes builtin Python types (e.g. `int` and `float`) into PyTorch data types
when these are passed as argument, instead of used as functions.

In summary, we:

- Implement `BuiltinVariable.as_proxy`, mapping Python types into PyTorch data types

cc @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @aakhundov

### CODE REVIEW

File: torch/_dynamo/variables/builtin.py [hunk 1]
```diff
@@ -409,6 +409,15 @@ def python_type(self):
     def as_python_constant(self):
         return self.fn
 
+    def as_proxy(self):
+        DTYPE = {
+            int: torch.int64,
```
Comment (lezcano) [inline torch/_dynamo/variables/builtin.py:414]: bool and complex?

Also, is this something we already do between the Python and the C++ code in PyTorch eager? If so, it'd be good to have a look at where are we doing that, to see if there's any other logic we may need to implement.

Comment (ysiraichi) [reply torch/_dynamo/variables/builtin.py:414]: Ah, forgot about `bool`. But, `complex` doesn't really work on eager. Should we enable it on dynamo?

Comment (ysiraichi) [reply torch/_dynamo/variables/builtin.py:414]: I think [this](https://github.com/pytorch/pytorch/blob/cb9a4fbbf25e5f15f862d11e206a4028e9898515/torch/csrc/utils/python_arg_parser.h#L730) is the function we are looking for. Adding `bool` seems to do the job.

**Review by ezyang — APPROVED** [review]

File: torch/_dynamo/variables/builtin.py (hunk 1 above)
Comment (lezcano) [reply torch/_dynamo/variables/builtin.py:415]: Thank you for digging this up!

**Review by lezcano — APPROVED** [review]

**Comment by OrenLeung** [comment]
thanks for this PR <3

**Pull request closed without merging**